app/classifier/tasks/highlighter/index.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import { Markdown } from 'markdownz';
+import GenericTaskEditor from '../generic-editor';
+import GenericTask from '../generic';
+import LabelEditor from './label-editor';
+import LabelRenderer from './label-renderer';
+import HighlighterSummary from './summary';
+import TextRenderer from '../../annotation-renderer/text';
+
+export default class Highlighter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.createButtons = this.createButtons.bind(this);
+    this.createLabelAnnotation = this.createLabelAnnotation.bind(this);
+    this.selectableArea = this.selectableArea.bind(this);
+  }
+
+  handleChange(toolIndex, e) {
+    let selection = document.getSelection();
+    this.createLabelAnnotation(selection, toolIndex);
+  }
+
+  createLabelAnnotation(selection, toolIndex) {
+    // currently we only deal with one selection at a time
+    const range = selection.getRangeAt(0);
+    const offset = this.getOffset(selection);
+    const start = offset + range.startOffset;
+    const end = offset + range.endOffset;
```

Comment:
I think I have a solution for this issue that substitutes newline characters with break tags in the LabelRenderer.